### PR TITLE
Enable nullability in some ToolStrip members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some ToolStrip members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8691)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 1]
```diff
@@ -3260,12 +3260,12 @@ internal virtual bool ProcessArrowKey(Keys keyCode)
         /// </summary>
         private bool ProcessLeftRightArrowKey(bool right)
         {
-            ToolStripItem selectedItem = GetSelectedItem();
+            ToolStripItem? selectedItem = GetSelectedItem();
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:3263]: This doesn't seem like it is being used. We can remove this line.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs [hunk 2]
```diff
@@ -3610,7 +3608,7 @@ protected override void OnMouseMove(MouseEventArgs mea)
 
                 // Fire mouse move on the item
                 // Transpose this to "client coordinates" of the ToolStripItem.
-                Point itemRelativePoint = item.TranslatePoint(new Point(mea.X, mea.Y), ToolStripPointType.ToolStripCoords, ToolStripPointType.ToolStripItemCoords);
+                Point itemRelativePoint = item!.TranslatePoint(new Point(mea.X, mea.Y), ToolStripPointType.ToolStripCoords, ToolStripPointType.ToolStripItemCoords);
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:3611]: Do you get complaints without this `!`?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:3611]: Yes, there is a warning even though we have a null check above.

Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStrip.cs:3610]: Hm, that is so strange 🤔

**Review by lonitra — APPROVED** [review]
🥳

**Comment by gpetrou** [comment]
@lonitra it looks like there was a different error now. Could you run the failed build again?

**Comment by lonitra** [comment]
Captured:
```
    System.Windows.Forms.Tests.ApplicationTests.Application_ScaleDefaultFont_Should_Cap_Rescale_ToTextScaleFactor225 [FAIL]
      Assert.Null() Failure
      Expected: (null)
      Actual:   [Font: Name=Times New Roman, Size=12, Units=3, GdiCharSet=1, GdiVerticalFont=False]
      Stack Trace:
           at CallSite.Target(Closure, CallSite, Type, Object)
           at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ApplicationTests.cs(291,0): at System.Windows.Forms.Tests.ApplicationTests.Application_ScaleDefaultFont_Should_Cap_Rescale_ToTextScaleFactor225()
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
           at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

**Pull request merged**